File: app/src/main/java/com/example/client/screens/main/activity/IMainView.java
package com.example.client.screens.main.activity;

import com.example.client.models.order.OrderModel;

public interface IMainView {
    void showHomeScreen();
    void showNotiScreen();
    void showProfileScreen();
    void showCart(int quantity);
    void hideCart();
    void showLoading();
    void hideLoading();
    void showErrorMessage(int errMessage);
    void showOrderCount(int count);
    void hideOrderCount();
    void showOrder(OrderModel order);
    void hideOrder();
}

File: app/src/main/java/com/example/client/screens/main/activity/IMainView.kt
package com.example.client.screens.main.activity

import android.content.Intent
import com.example.client.base.IBaseView
import com.example.client.models.order.OrderModel
import com.example.client.screens.branch.BranchActivity

interface IMainView: IBaseView {
    fun showCart(quantity: Int)
    fun hideCart()
    fun showErrorMessage(errMessage: Int)
    fun showOrderCount(count: Int)
    fun hideOrderCount()
    fun showOrder(order: OrderModel)
    fun hideOrder()
    fun navigateToOrderDetail(orderCode: String)
    fun toBranchScreen()
}